File: 프로그래머스/unrated/181900. 글자 지우기/글자 지우기.java
//1. 일반 풀이

// class Solution {
//     public String solution(String my_string, int[] indices) {
       
//         char[] ch = my_string.toCharArray();
//         StringBuilder sb = new StringBuilder();
        
//         for(int i : indices)
//             ch[i]='-';
        
//         for(char c : ch)
//             if(c!='-') 
//                 sb.append(c);
        
//         return sb.toString();
//     }
// }

//2. 스트림 풀이
import java.util.stream.IntStream;
import java.util.stream.Collectors;
class Solution{
    public String solution(String my_string, int[] indices) {
     return IntStream.range(0, my_string.length()).mapToObj(i->(IntStream.of(indices).anyMatch(x-> x == i) ? "-" : my_string.charAt(i))).map(String::valueOf).collect(Collectors.joining()).replaceAll("-","");
    }
}

File: 프로그래머스/0/181900. 글자 지우기/글자 지우기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string my_string, vector<int> indices) {
    string answer = "";
    for(int i = 0; i < indices.size(); i++){
        my_string[indices[i]] = ' ';
    }
    for(char c : my_string){
        if(c == ' ') continue;
        answer += c;
    }
    return answer;
}